src/go/rpk/pkg/api/admin/api_topic_recovery.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// RecoveryRequestParams represents the request body schema for the automated recovery API endpoint.
+type RecoveryRequestParams struct {
+	TopicNamesPattern string `json:"topic_names_pattern"`
+	RetentionBytes    int    `json:"retention_bytes"`
+	RetentionMs       int    `json:"retention_ms"`
+}
+
+type RecoveryStartResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+// TopicDownloadCounts represents the count of downloads for a topic.
+type TopicDownloadCounts struct {
+	TopicNamespace      string `json:"topic_namespace"`
+	PendingDownloads    int    `json:"pending_downloads"`
+	SuccessfulDownloads int    `json:"successful_downloads"`
+	FailedDownloads     int    `json:"failed_downloads"`
+}
+
+// TopicRecoveryStatus represents the status of the automated recovery for a topic.
+type TopicRecoveryStatus struct {
+	State           string                `json:"state"`
+	TopicDownloads  []TopicDownloadCounts `json:"topic_download_counts"`
+	RecoveryRequest RecoveryRequestParams `json:"request"`
+}
+
+// StartAutomatedRecovery starts the automated recovery process by sending a request to the automated recovery API endpoint.
+func (a *AdminAPI) StartAutomatedRecovery(ctx context.Context) error {
+	requestParams := &RecoveryRequestParams{
+		TopicNamesPattern: ".*",
+	}
+	var response RecoveryStartResponse
+	a.sendAny(ctx, http.MethodPost, "/v1/cloud_storage/automated_recovery", requestParams, &response)
+
+	return nil
+}
+
+// PollAutomatedRecoveryStatus polls the automated recovery status API endpoint to retrieve the latest status of the recovery process.
+func (a *AdminAPI) PollAutomatedRecoveryStatus(ctx context.Context) (*TopicRecoveryStatus, error) {
+	var response TopicRecoveryStatus
+	return &response, a.sendAny(ctx, http.MethodGet, "/v1/cloud_storage/automated_recovery", http.NoBody, &response)
+}
+
+// WaitUntilAutomatedRecoveryFinished waits until the automated recovery process for the topic is finished.
+// TODO: Add a timeout or max retries.
+func (a *AdminAPI) WaitUntilAutomatedRecoveryFinished(ctx context.Context, pollingInterval time.Duration) error {
```

Comment:
I don't think this should be a method of the AdminAPI but instead be a function defined in the `cmd/topic/autorestore` package.

Reply:
I can see that the way I've implemented it here runs counter to how for instance `rpk cluster maintenance enable` is implemented

The reason I put it in this AdminAPI is that we'll probably want to use this method in the cloud agent at a later point, so placing the logic in the AdminAPI would prevent having to duplicate the logic in both the agent and rpk

Is moving the logic to the `cmd/topic/autorestore` a hard requirement?

Reply:
I won't say "hard" because it sounds harsh :sweat_smile: but what I see is:

The API/admin package is publicly available and probably used by others, so it's important to maintain backward compatibility. Any hard-coded internal logic should not be included in this package and instead should be implemented by the users of this package, even if that means duplicating some logic in other parts of the code.

The reason for keeping the logic outside of the package is to allow for flexibility in the future. The logic may need to change (for the agent) or new parameters may need to be added (or even logging), which would require a change to the public API. This could potentially break the code for unknown users. By keeping the logic outside of the package, changes can be made to the internal code without affecting others.